x-pack/plugins/monitoring/server/alerts/cpu_usage_rule.ts

```diff
@@ -62,28 +63,51 @@ export class CpuUsageRule extends BaseRule {
     const duration = parseDuration(params.duration);
     const endMs = +new Date();
     const startMs = endMs - duration;
+    const filterQuery = params.filterQuery
+      ? (JSON.parse(params.filterQuery) as QueryDslQueryContainer)
+      : undefined;
+
     const stats = await fetchCpuUsageNodeStats(
-      esClient,
-      clusters,
-      startMs,
-      endMs,
-      Globals.app.config.ui.max_bucket_size,
-      params.filterQuery
+      {
+        esClient,
+        clusterUuids: clusters.map((cluster) => cluster.clusterUuid),
+        startMs,
+        endMs,
+        filterQuery,
+        logger: this.scopedLogger,
+      },
+      Globals.app.config
     );
-    return stats.map((stat) => {
-      if (Globals.app.config.ui.container.elasticsearch.enabled) {
-        stat.cpuUsage =
-          (stat.containerUsage / (stat.containerPeriods * stat.containerQuota * 1000)) * 100;
+
+    return stats.map((stat) => ({
+      clusterUuid: stat.clusterUuid,
+      ...this.outcomeAndSeverity(stat, params.threshold!),
+      meta: {
+        ...stat,
+        threshold: params.threshold!,
+      },
+      ccs: stat.ccs,
+    }));
+  }
+
+  private outcomeAndSeverity(
+    stat: AlertCpuUsageNodeStats,
+    threshold: number
+  ): { shouldFire: boolean; severity: AlertSeverity } {
+    if (
+      stat.missingLimits ||
+      stat.limitsChanged ||
+      stat.unexpectedLimits ||
+      stat.cpuUsage === undefined
+    ) {
+      let severity = AlertSeverity.Warning;
+      if (stat.cpuUsage && stat.cpuUsage > threshold) {
```

Comment:
I do need the check, in the case that Kibana is configured for a container workload but it doesn't have limits set, we'll fallback to the normal CPU metric which is always reported. And in that case, if that value is above the threshold I want the alert to be danger, not warning. But it might not matter a million. I'm leaving it as it is for now!